### DISCUSSION TITLE
Coupling Crop model and CWatM

### DISCUSSION THREAD

**Discussion by GautameeBaviskar** [iiasa/CWatM#83] [Q&A] [answered]
Hi, I am Gautamee, currently doing my PhD research on water demand and footprint assessment. I was curious to know if there have been any studies or attempts made to couple crop models like AquaCrop and CWatM? If you have come across any research on this topic, please share the references.

**Comment by mikhailsmilovic** [comment]
Guten Tag Gautamee, with have integrated crop-specific water use with the CROPWAT method. It is similar to Aquacrop, only that it estimates relative yield instead of actual yield. However, for questions of water demand and footprint assessment, the CROPWAT method will work well. We are currently preparing a paper describing this and a YouTube tutorial for September. 
We are currently preparing a project to couple PyAEZ (https://github.com/gicait/PyAEZ) with CWatM to estimate actual yield -- this we expect to have by the end of 2024. 

**Description of crop-specific water use in CWatM:**

Crop water use and relative yields are implemented following the methodology of stage-specific crop coefficients and yield sensitivities from FAO Irrigation and drainage papers 24, 56, 66. Crop choices and related parameters are user inputs. Crop parameters include planting month, length of four growth stages, growth-stage specific relative water uses (crop coefficients), and yield sensitivities. These parameters can be informed by crop- and region-specific studies, FAO Irrigation and drainage papers, FAO crop information, and general studies referencing the crop-coefficient methodology, including GAEZ model documentation.

The required inputs for each crop include the following:

- 	Planting calendar month,
- 	Four growth-stage lengths,
- 	Stage-specific crop coefficients (Kc_crop) scaling reference evapotranspiration to determine potential evapotranspiration,
- 	Stage-specific yield sensitivities (Ky_crop) relating deficit to stage-specific sensitivities, and
- 	Maps of irrigated and non-irrigated fractional crop areas.

On the first day of the month, crops with the final growth stage completed are harvested, and crops with the relevant planting month are sown, assuming sufficient available land. All crop inputs except the maps are included using an Excel Sheet, as shown in Figure S1. Crops are planted in the order they are given in the Sheet.

**Comment by mikhailsmilovic** [comment, marked as answer]
We have now implemented crops using an FAO/AEZ-inspired crop growth-stage approach. The following two videos explain how to prepare maps and include crops for the Danube basin. 
1. https://youtu.be/6pBaK2qRliE?si=HffDwOMquBkFnjYg
2. https://youtu.be/YvdVTB4wtTA?si=00dXzuDmeACy11mf

We are currently preparing the model description paper. Coupling with different crop models, including Aquacrop, is also possible and interesting. Below are the instructions to include crops

---

In the settings text file, in the section [OPTIONS], the following lines are to be included:

includeCrops = True
use_GeneralCropnonIrr = 0
use_GeneralCropIrr = 0
activate_fallow = 1
leftoverIrrigatedCropIsRainfed = 0
automaticFallowingIrr = True
moveIrrFallowToNonIrr = False

In the section [FILE_PATHS], include:
Excel_settings_file = $(PathMaps)\settings\cwatm_settings.xlsx

#### For each crop included in the Excel_settings_file, CWatM looks for two maps: one for the irrigated fraction and one for the non-irrigated fraction.
Include paths to these maps as follows in [LANDCOVER]:

NAME1_Irr = _path to map irrigated NAME1 fraction_
NAME2_Irr  = _path to map irrigated NAME2 fraction_
NAME1_nonIrr = _path to map non-irrigated NAME1 fraction_
NAME2_nonIrr = _path to map non-irrigated NAME2 fraction_
...
**for example:**
Cotton_Irr = $(FILE_PATHS:PathMaps)/landcover/crops/Cotton_Irr_totalend.nc
Wheat_Irr = $(FILE_PATHS:PathMaps)/landcover/crops/Wheat_Irr_totalend.nc
Cotton_nonIrr = $(FILE_PATHS:PathMaps)/landcover/crops/Cotton_nonIrr_totalend.nc
Wheat_nonIrr = $(FILE_PATHS:PathMaps)/landcover/crops/Wheat_nonIrr_totalend.nc

The Excel requires the planting month for each crop, the length of four separate growth stages (in months), static crop coefficients for each growth stage, and, optionally, the yield sensitivity for each growth stage. If a study or report shows only three crop coefficients or crop coefficients that change throughout a growth stage, these must be adapted to four static growth-stage-specific crop coefficients. 
[cwatm_settings.xlsx](https://github.com/iiasa/CWatM/files/15067792/cwatm_settings.xlsx)

Crop coefficients are available for many crops with the FAO:
https://www.fao.org/land-water/databases-and-software/crop-information/en/

The manual for the model GAEZ version 4 has crop coefficients for three growth stages in https://www.fao.org/3/cb4744en/cb4744en.pdf

We are developing a variation on the crop method implemented, with three growth stages and Julian day inputs.
